common/persistence/persistenceInterface.go

```diff
@@ -799,8 +735,17 @@ func ProtoWorkflowExecutionToPartialInternalExecution(info *persistenceblobs.Wor
 		AutoResetPoints:                        info.GetAutoResetPoints(),
 	}
 
+	// Back compat for GetHistorySize
+	if info.GetHistorySize() >= 0 && info.GetExecutionStats() == nil {
+		executionInfo.ExecutionStats = &persistenceblobs.ExecutionStats{HistorySize: 0}
```

Comment:
Ignore bug here - fixed in a separate branch/commit with a test to ensure this works correctly migration. Will separate and apply to this PR shortly.